inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,500 @@
+    std::vector<Type> catch_numbers_at_age;      /*!< Expected values???: Catch in numbers at age*/
+    std::vector<Type> catch_weight_at_age;       /*!< Expected values???: Weight at age for catch*/
+    std::vector<Type> expected_catch;            /*!< Expected values: Catch*/
+    std::vector<Type> expected_index;            /*!< Expected values: Index (CPUE)*/
+
+    /// recruitment
+    int recruitment_id = -999;                                /*!< id of recruitment model object*/
+    std::shared_ptr<fims::RecruitmentBase<Type> > recruitment; /*!< shared pointer to recruitment module */
+
+    // growth
+    int growth_id = -999;                           /*!< id of growth model object*/
+    std::shared_ptr<fims::GrowthBase<Type> > growth; /*!< shared pointer to growth module */
+
+    // maturity
+    int maturity_id = -999;                             /*!< id of maturity model object*/
+    std::shared_ptr<fims::MaturityBase<Type> > maturity; /*!< shared pointer to maturity module */
+
+    // fleet
+    int fleet_id = -999;                                    /*!< id of fleet model object*/
+    std::vector<std::shared_ptr<fims::Fleet<Type> > > fleets; /*!< shared pointer to fleet module */
+
+    // this -> means you're referring to a class member (member of self)
+
+    // std::vector<std::shared_ptr<fims::Fleet<Type>> > surveys;
+    Population()
+    {
+      this->id = Population::id_g++;
+    }
+
+    /**
+     * @brief Initialize values. Called once at the start of model run.
+     *
+     * @param index_ya dimension folded index for year and age
+     * @param age age index
+     */
+    void Initialize(int nyears, int nseasons, int nages)
+    {
+      // size all the vectors to length of nages
+      nfleets = fleets.size();
+      ages.resize(nages);
+      catch_numbers_at_age.resize(nyears * nages * nfleets);
+      mortality_F.resize(nyears * nages);
+      mortality_Z.resize(nyears * nages);
+      proportion_mature_at_age.resize(nyears * nages);
+      weight_at_age.resize(nages);
+      catch_weight_at_age.resize(nyears * nages * nfleets);
+      unfished_numbers_at_age.resize((nyears + 1) * nages);
+      numbers_at_age.resize((nyears + 1) * nages);
+      expected_catch.resize(nyears * nfleets);
+      expected_index.resize(nyears * nfleets);
+      biomass.resize((nyears + 1));
+      unfished_spawning_biomass.resize((nyears + 1));
+      spawning_biomass.resize((nyears + 1));
+      log_naa.resize(nages);
+      log_Fmort.resize(nfleets * nyears);
+      log_M.resize(nyears * nages);
+      log_q.resize(nfleets);
+      naa.resize(nages);
+      Fmort.resize(nfleets * nyears);
+      M.resize(nyears * nages);
+      q.resize(nfleets);
+    }
+
+    /**
+     * @brief Prepare to run the population loop. Called at each model itartion, and used
+     *  to zero out derived quantities, values that were summed, etc.
+     *
+     */
+    void Prepare()
```

Comment:
Under a scenario where there is one fishing fleet operating on two populations, would initializing the fleet module within the population Prepare function initialize the fleet twice, for each population? Tagging @msupernaw